### PR TITLE
partition: add doc for global index

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/18543
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: placement-rules-in-sql.md [hunk 1]
```diff
@@ -295,12 +297,13 @@ PARTITION BY RANGE( YEAR(purchased) ) (
...
 
-如果没有为表中的某个分区指定任何放置策略，该分区将尝试继承表上可能存在的策略。在上面示例中：
+如果没有为表中的某个分区指定任何放置策略，该分区将尝试继承表上可能存在的策略。如果表有[全局索引](/partitioned-table.md#全局索引)，索引将应用与表相同的放置策略。在上面示例中：
 
 - `p0` 分区将会应用 `storageforhisotrydata` 策略
 - `p4` 分区将会应用 `storagefornewdata` 策略
 - `p1`、`p2`、`p3` 分区将会应用表 `t1` 的放置策略 `companystandardpolicy`
-- 如果 `t1` 没有绑定任何策略，`p1`、`p2`、`p3` 会继承数据库或全局的默认策略
+- 全局索引 `idx` 将应用与表 `t1` 相同的 `companystandardpolicy` 放置策略
+- 如果没有为表 `t1`、`p1`、`p2` 和 `p3` 分区指定放置策略，全局索引 `idx` 将继承数据库默认策略或全局默认策略
```
Comment (Defined2014) [inline placement-rules-in-sql.md:306]:
```suggestion
- 如果没有为表 `t1` 指定放置策略，`p1`、`p2` 和 `p3` 分区以及全局索引 `idx` 将继承数据库默认策略或全局默认策略
```

File: partitioned-table.md [hunk 2]
```diff
@@ -1666,6 +1664,66 @@ CREATE TABLE t (a varchar(20), b blob,
...
+    col2 DATE NOT NULL,
+    PRIMARY KEY (col2) clustered
+) PARTITION BY HASH(col1) PARTITIONS 5;
+```
+
+```
+ERROR 1503 (HY000): A CLUSTERED INDEX must include all columns in the table's partitioning function
+```
+
+聚簇索引不能成为全局索引，是因为如果聚簇索引是全局索引，则表将不再分区。这是因为聚类索引的键也是记录键，应该位于分区级别上，但全局索引位于表级别，这就造成了冲突。
```
Comment (Defined2014) [inline partitioned-table.md:1708]:
```suggestion
聚簇索引不能成为全局索引，是因为如果聚簇索引是全局索引，则表将不再分区。这是因为聚簇索引的键也是行数据的键，而行数据应该是分区级别的，但全局索引是表级别，这就造成了冲突。
```

**Review by Defined2014 — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -1919,13 +1919,17 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+> 该变量控制的功能为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
 - 类型：布尔型
 - 默认值：`OFF`
 - 可选值：`OFF`，`ON`
-- 这个变量用于控制是否支持对分区表创建 `Global index`。`Global index` 当前正处于开发阶段，**不推荐修改该变量值**。
+- 该变量控制是否支持为分区表创建 [`全局索引`](/partitioned-table.md#全局索引)。启用此变量后，TiDB 将创建唯一索引，该索引不包含分区表达式中的所有列。
```
Comment (hfxsd) [inline system-variables.md:1932]:
```suggestion
- 该变量控制是否支持为分区表创建 [全局索引](/partitioned-table.md#全局索引)。启用此变量后，TiDB 将创建唯一索引，该索引不包含分区表达式中的所有列。
```

File: system-variables.md [hunk 4]
```diff
@@ -1919,13 +1919,17 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+> 该变量控制的功能为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
 - 类型：布尔型
 - 默认值：`OFF`
 - 可选值：`OFF`，`ON`
-- 这个变量用于控制是否支持对分区表创建 `Global index`。`Global index` 当前正处于开发阶段，**不推荐修改该变量值**。
+- 该变量控制是否支持为分区表创建 [全局索引](/partitioned-table.md#全局索引)。启用此变量后，TiDB 将创建唯一索引，该索引不包含分区表达式中的所有列。
```
Comment (hfxsd) [inline system-variables.md:1932]:
```suggestion
- 该变量控制是否支持为分区表创建[全局索引](/partitioned-table.md#全局索引)。启用此变量后，TiDB 将创建唯一索引，该索引不包含分区表达式中的所有列。
```

File: partitioned-table.md [hunk 5]
```diff
@@ -1666,6 +1664,66 @@ CREATE TABLE t (a varchar(20), b blob,
 ERROR 1503 (HY000): A UNIQUE INDEX must include all columns in the table's partitioning function
 ```
 
+#### 全局索引
+
+如果你需要创建的唯一索引**不包含分区表达式中使用的所有列**，可以通过启用 [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-从-v760-版本开始引入) 系统变量来实现。
+
+以前，分区表上的索引是为每个分区创建的，因此有一个限制，即所有唯一键都需要包含所有分区列。唯一性只能在每个分区内强制执行。全局索引将在表级别创建，因此无论分区如何，它都可以强制执行唯一性。注意，全局索引对分区管理有影响，`DROP`、`TRUNCATE` 和 `REORGANIZE PARTITION` 也需要管理表级全局索引。
```
Comment (hfxsd) [inline partitioned-table.md:1671]:
```suggestion
以前，分区表上的索引是为每个分区创建的，因此有一个限制，即所有唯一键都需要包含所有分区键。唯一性只能在每个分区内强制执行。全局索引将在表级别创建，因此无论分区如何，它都可以强制执行唯一性。注意，全局索引对分区管理有影响，`DROP`、`TRUNCATE` 和 `REORGANIZE PARTITION` 也需要管理表级全局索引。
```

File: partitioned-table.md [hunk 6]
```diff
@@ -1666,6 +1664,92 @@ CREATE TABLE t (a varchar(20), b blob,
 ERROR 1503 (HY000): A UNIQUE INDEX must include all columns in the table's partitioning function
 ```
 
+#### 全局索引
+
+如果你需要创建的唯一索引**不包含分区表达式中使用的所有列**，可以通过启用 [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-从-v760-版本开始引入) 系统变量来实现。
+
+以前，分区表上的索引是为每个分区创建的，因此有一个[限制](#分区键主键和唯一键)，即分区表的每个唯一键，必须包含分区表达式中用到的所有列。唯一性只能在每个分区内强制实现。全局索引将在表级别创建，因此无论分区如何，它都可以强制实现唯一性。注意，全局索引对分区管理有影响，`DROP`、`TRUNCATE` 和 `REORGANIZE PARTITION` 也需要管理表级全局索引。
+
+启用此变量后，任何不符合前述约束条件的唯一索引都需要使用 `GLOBAL` 属性，然后成为全局索引。
```
Comment (Defined2014) [inline partitioned-table.md:1673]:
```suggestion
启用此变量后，任何不符合前述约束条件的唯一索引都需要使用 `GLOBAL` 属性来将其创建为全局索引。
```

File: sql-statements/sql-statement-add-column.md [hunk 7]
```diff
@@ -136,6 +136,7 @@ SELECT * FROM t1;
 * 不支持将新添加的列设为 `PRIMARY KEY`。
 * 不支持将新添加的列设为 `AUTO_INCREMENT`。
 * 对添加生成列有局限性，具体可参考：[生成列局限性](/generated-columns.md#生成列的局限性)。
+* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展，你可以将 `GLOBAL` ColumnOption 设置为 `KEY` 来添加新列。该扩展与 MySQL 不兼容。
```
Comment (Defined2014) [inline sql-statements/sql-statement-add-column.md:139]:
```suggestion
* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展，你可以在创建新列时通过将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为 `GLOBAL` 来设置全局索引。该扩展与 MySQL 不兼容。
```

File: sql-statements/sql-statement-add-index.md [hunk 8]
```diff
@@ -118,6 +120,7 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 * TiDB 支持解析 `FULLTEXT` 语法，但不支持使用 `FULLTEXT` 索引。
 * 不支持降序索引（类似于 MySQL 5.7）。
 * 无法向表中添加 `CLUSTERED` 类型的 `PRIMARY KEY`。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
+* TiDB 对[全局索引](/partitioned-table.md#global-indexes)进行了扩展。你可以使用 `GLOBAL` IndexOption 添加 `PRIMARY KEY` 或 `UNIQUE INDEX` 。该扩展与 MySQL 不兼容。
```
Comment (Defined2014) [inline sql-statements/sql-statement-add-index.md:123]:
```suggestion
* TiDB 对[全局索引](/partitioned-table.md#global-indexes)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```

File: sql-statements/sql-statement-create-index.md [hunk 9]
```diff
@@ -407,6 +408,7 @@ CREATE UNIQUE INDEX c1 ON t1 (c1) INVISIBLE;
 * 表达式索引与视图存在兼容性问题。通过视图进行查询时，无法使用上表达式索引。
 * 表达式索引与 Binding 存在兼容性问题。当表达式索引中的表达式存在常量时，对应查询所建的 Binding 会扩大范围。假设表达式索引中的表达式为 `a+1`，对应的查询条件为 `a+1 > 2`。则建立的 Binding 为 `a+? > ?`，这会导致像 `a+2 > 2` 这样的查询也会强制使用表达式索引，得到一个较差的执行计划。这同样影响 SQL Plan Management (SPM) 中的捕获和演进功能。
 * 多值索引写入的数据必须与定义类型完全匹配，否则数据写入失败。详见[创建多值索引](/sql-statements/sql-statement-create-index.md#创建多值索引)。
+* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展。你可以使用 `GLOBAL` IndexOption 创建 `UNIQUE` `KEY`。该扩展与 MySQL 不兼容。
```
Comment (Defined2014) [inline sql-statements/sql-statement-create-index.md:411]:
```suggestion
* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```

File: sql-statements/sql-statement-create-table.md [hunk 10]
```diff
@@ -234,6 +235,7 @@ mysql> DESC t1;
 * `COMMENT` 属性不支持 `WITH PARSER` 选项。
 * TiDB 在单个表中默认支持 1017 列，最大可支持 4096 列。InnoDB 中相应的数量限制为 1017 列，MySQL 中的硬限制为 4096 列。详情参阅 [TiDB 使用限制](/tidb-limitations.md)。
 * 当前仅支持 Range、Hash 和 Range Columns（单列）类型的分区表，详情参阅[分区表](/partitioned-table.md)。
+* TiDB 对[全局索引](/partitioned-table.md#global-indexes)进行了扩展。你可以使用 `GLOBAL` IndexOption 创建 `PRIMARY KEY` 或 `UNIQUE INDEX`。该扩展与 MySQL 不兼容。
```
Comment (Defined2014) [inline sql-statements/sql-statement-create-table.md:238]:
```suggestion
* TiDB 对[全局索引](/partitioned-table.md#global-indexes)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```

**Review by Frank945946 — APPROVED** [review]

File: partitioned-table.md [hunk 11]
```diff
@@ -1666,6 +1664,92 @@ CREATE TABLE t (a varchar(20), b blob,
...
++--------------+------------+------------+----------+--------------+-------------+----------+---------------+------------+----------+------------+-----------+-----------+
+3 rows in set (0.00 sec)
+```
+
+在分区非分区表、或重新分区已经分区的表时，可以将索引更新为全局索引或默认本地索引。
+
+```sql
+ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCAL, uidx3 GLOBAL);
+```
+
```
Comment (Frank945946) [inline partitioned-table.md:1752]:
```suggestion
### 全局索引的限制
-如果未显式指定 GLOBAL 关键字，将默认创建局部索引（local Index）。
-GLOBAL 和 LOCAL 关键字仅适用于分区表，对非分区表没有影响。即在非分区表中，全局索引和局部索引之间没有区别。
-DDL 操作如 ADD PARTITION、DROP PARTITION、TRUNCATE PARTITION、REORGANIZE PARTITION、SPLIT PARTITION 和 EXCHANGE PARTITION 等 DDL 会触发全局索引的更新，DDL 执行结果将在全局索引更新完成后才会返回。这会影响需要快速完成 DDL 的场景，例如数据归档操作，EXCHANGE PARTITION、TRUNCATE PARTITION 和 DROP PARTITION。
-默认情况下，分区表的主键（PK）为聚簇索引，必须包含分区键。如果要求主键不包含分区建，可以在建表时显式指定主键为非聚簇的全局索引，例如：PRIMARY KEY(col1, col2) NONCLUSTERED GLOBAL。

```

**Review by Frank945946 — APPROVED** [review]

File: partitioned-table.md (hunk 6 above)
Comment (Frank945946) [inline partitioned-table.md:1673]:
```suggestion
以前，分区表上的索引是为每个分区创建的，因此有一个[限制](#分区键主键和唯一键)，即分区表的每个唯一键，必须包含分区表达式中用到的所有列。唯一性只能在每个分区内强制实现。全局索引将在表级别创建，因此无论分区如何，它都可以强制实现唯一性。
如果你需要创建的唯一索引**不包含分区表达式中使用的所有列**，可以通过启用 [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-从-v760-版本开始引入) 系统变量来实现。
启用此变量后，任何不符合前述约束条件的唯一索引都需要使用 `GLOBAL` 属性来将其创建为全局索引。
注意，全局索引对分区管理有影响，`DROP`、`TRUNCATE` 和 `REORGANIZE PARTITION` 也需要管理表级全局索引，即这些 DDL 需要更新完全局索引后才会返回结果。
```

**Review by Frank945946 — APPROVED** [review]

File: partitioned-table.md [hunk 12]
```diff
@@ -1666,6 +1664,102 @@ CREATE TABLE t (a varchar(20), b blob,
 ERROR 1503 (HY000): A UNIQUE INDEX must include all columns in the table's partitioning function
 ```
 
+#### 全局索引
+
+在引入全局索引 (Global Index) 之前，TiDB 为每个分区创建一个局部索引 (Local Index)，即一个分区对应一个局部索引。因此，主键和唯一键必须包含分区键，以确保数据的全局唯一性。此外，当查询的数据跨越多个分区时，系统需要扫描每个分区的数据才能返回结果。
+
+为解决这些问题，TiDB 从 v8.3.0 开始引入全局索引。全局索引通过一个单一的索引覆盖整个表的数据，使得主键和唯一键在不包含分区键的情况下仍能保持全局唯一性。同时，全局索引可以一次性访问跨多个分区的数据，显著提升了针对非分区键的查询性能。
+
+以前，分区表上的索引是为每个分区创建的，因此有一个[限制](#分区键主键和唯一键)，即分区表的每个唯一键，必须包含分区表达式中用到的所有列。唯一性只能在每个分区内强制实现。全局索引将在表级别创建，因此无论分区如何，它都可以强制实现唯一性。
```
Comment (hfxsd) [inline partitioned-table.md:1673]:
```suggestion
在引入全局索引 (Global Index) 之前，TiDB 为每个分区创建一个局部索引 (Local Index)，即一个分区对应一个局部索引。因此有一个[限制](#分区键主键和唯一键)，主键和唯一键必须包含分区键，以确保数据的全局唯一性。此外，当查询的数据跨越多个分区时，系统需要扫描每个分区的数据才能返回结果。

为解决这些问题，TiDB 从 v8.3.0 开始引入全局索引。全局索引通过一个单一的索引覆盖整个表的数据，使得主键和唯一键在不包含分区键的情况下仍能保持全局唯一性。同时，全局索引可以一次性访问跨多个分区的数据，显著提升了针对非分区键的查询性能。
```

Comment (Frank945946) [reply partitioned-table.md:1673]: LGTM

File: partitioned-table.md [hunk 13]
```diff
@@ -1672,6 +1670,100 @@ CREATE TABLE t (a varchar(20), b blob,
...
++--------------+------------+------------+----------+--------------+-------------+----------+---------------+------------+----------+------------+-----------+-----------+
+3 rows in set (0.00 sec)
+```
+
+在分区非分区表、或重新分区已经分区的表时，可以将索引更新为全局索引或默认本地索引。
+
+```sql
+ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCAL, uidx3 GLOBAL);
+```
+##### 全局索引的限制
```
Comment (hfxsd) [inline partitioned-table.md:1760]:
```suggestion

##### 全局索引的限制
```

File: sql-statements/sql-statement-add-index.md [hunk 14]
```diff
@@ -118,6 +120,7 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 * TiDB 支持解析 `FULLTEXT` 语法，但不支持使用 `FULLTEXT` 索引。
 * 不支持降序索引（类似于 MySQL 5.7）。
 * 无法向表中添加 `CLUSTERED` 类型的 `PRIMARY KEY`。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
+* TiDB 对[全局索引](/partitioned-table.md#global-indexes)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```
Comment (hfxsd) [inline sql-statements/sql-statement-add-index.md:123]:
```suggestion
* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```

File: sql-statements/sql-statement-create-table.md [hunk 15]
```diff
@@ -234,6 +235,7 @@ mysql> DESC t1;
 * `COMMENT` 属性不支持 `WITH PARSER` 选项。
 * TiDB 在单个表中默认支持 1017 列，最大可支持 4096 列。InnoDB 中相应的数量限制为 1017 列，MySQL 中的硬限制为 4096 列。详情参阅 [TiDB 使用限制](/tidb-limitations.md)。
 * 分区表支持 `HASH`、`RANGE`、`LIST` 和 `KEY` [分区类型](/partitioned-table.md#分区类型)。对于不支持的分区类型，TiDB 会报 `Warning: Unsupported partition type %s, treat as normal table` 错误，其中 `%s` 为不支持的具体分区类型。
+* TiDB 对[全局索引](/partitioned-table.md#global-indexes)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```
Comment (hfxsd) [inline sql-statements/sql-statement-create-table.md:238]:
```suggestion
* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```

File: partitioned-table.md [hunk 16]
```diff
@@ -1672,6 +1670,101 @@ CREATE TABLE t (a varchar(20), b blob,
 ERROR 1503 (HY000): A UNIQUE INDEX must include all columns in the table's partitioning function
 ```
 
+#### 全局索引
+
```
Comment (qiancai) [inline partitioned-table.md:1674]: 这里要加

File: partitioned-table.md [hunk 17]
```diff
@@ -1461,6 +1461,10 @@ SELECT store_id, COUNT(department_id) AS c
 
 本节讨论分区键，主键和唯一键之间的关系。一句话总结它们之间的关系要满足的规则：**分区表的每个唯一键，必须包含分区表达式中用到的所有列**。
 
+> **注意：**
+>
+> 该规则仅适用于 [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-new-in-v760) 未开启的场景。
```
Comment (hfxsd) [inline partitioned-table.md:1466]:
```suggestion
> 该规则仅适用于 [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-从-v760-版本开始引入) 未开启的场景。
```

File: partitioned-table.md [hunk 18]
```diff
@@ -1672,6 +1674,105 @@ CREATE TABLE t (a varchar(20), b blob,
...
+
+为解决这些问题，TiDB 从 v8.3.0 开始引入全局索引。全局索引通过一个单一的索引覆盖整个表的数据，使得主键和唯一键在不包含分区键的情况下仍能保持全局唯一性。同时，全局索引可以一次性访问跨多个分区的数据，显著提升了针对非分区键的查询性能。
+
+> **警告：**
+>
+> 该变量控制的功能为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+如果你需要创建的唯一索引**不包含分区表达式中使用的所有列**，可以通过启用 [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-从-v760-版本开始引入) 系统变量来实现。启用此变量后，任何不符合前述约束条件的唯一索引都需要使用 `GLOBAL` 属性来将其创建为全局索引。
+
+注意，全局索引对分区管理有影响，`DROP`、`TRUNCATE` 和 `REORGANIZE PARTITION` 也需要管理表级全局索引，即这些 DDL 需要更新完全局索引后才会返回结果。
```
Comment (hfxsd) [inline partitioned-table.md:1689]:
```suggestion
> **注意：**
>
> 全局索引对分区管理有影响，`DROP`、`TRUNCATE` 和 `REORGANIZE PARTITION` 也需要管理表级全局索引，即这些 DDL 需要更新完全局索引后才会返回结果。
```

File: partitioned-table.md [hunk 19]
```diff
@@ -1672,6 +1674,107 @@ CREATE TABLE t (a varchar(20), b blob,
...
+) PARTITION BY HASH(col1) PARTITIONS 5;
+```
+
+```
+ERROR 1503 (HY000): A CLUSTERED INDEX must include all columns in the table's partitioning function
+```
+
+聚簇索引不能成为全局索引，是因为如果聚簇索引是全局索引，则表将不再分区。这是因为聚簇索引的键也是行数据的键，而行数据应该是分区级别的，但全局索引是表级别，这就造成了冲突。如果需要将主键设置为全局索引，则需要显式设置该主键为非聚簇索引，如 `PRIMARY KEY(col1, col2) NONCLUSTERED GLOBAL`。
+
+你可以通过 [`SHOW CREATE TABLE`](/sql-statements/sql-statement-show-create-table.md) 输出中的 `GLOBAL IndexOption` 来识别全局索引。
```
Comment (qiancai) [inline partitioned-table.md:1728]:
```suggestion
你可以通过 [`SHOW CREATE TABLE`](/sql-statements/sql-statement-show-create-table.md) 输出中的 `GLOBAL` 索引选项来识别全局索引。
```

File: sql-statements/sql-statement-add-column.md [hunk 20]
```diff
@@ -136,6 +136,7 @@ SELECT * FROM t1;
 * 不支持将新添加的列设为 `PRIMARY KEY`。
 * 不支持将新添加的列设为 `AUTO_INCREMENT`。
 * 对添加生成列有局限性，具体可参考：[生成列局限性](/generated-columns.md#生成列的局限性)。
+* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展，你可以在添加新列时通过将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为 `GLOBAL` 来设置全局索引。该扩展与 MySQL 不兼容。
```
Comment (qiancai) [inline sql-statements/sql-statement-add-column.md:139]:
```suggestion
* TiDB 对[分区表](/partitioned-table.md)进行了扩展，你可以在添加新列时通过将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为 `GLOBAL` 来设置[全局索引](/partitioned-table.md#全局索引)。该扩展与 MySQL 不兼容。
```

File: sql-statements/sql-statement-add-index.md [hunk 21]
```diff
@@ -118,6 +120,7 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 * TiDB 支持解析 `FULLTEXT` 语法，但不支持使用 `FULLTEXT` 索引。
 * 不支持降序索引（类似于 MySQL 5.7）。
 * 无法向表中添加 `CLUSTERED` 类型的 `PRIMARY KEY`。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
+* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```
Comment (qiancai) [inline sql-statements/sql-statement-add-index.md:123]:
```suggestion
* TiDB 对[分区表](/partitioned-table.md)进行了扩展。你可以指定 `GLOBAL` 索引选项将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为[全局索引](/partitioned-table.md#全局索引)。该扩展与 MySQL 不兼容。
```

File: sql-statements/sql-statement-create-index.md [hunk 22]
```diff
@@ -407,6 +408,7 @@ CREATE UNIQUE INDEX c1 ON t1 (c1) INVISIBLE;
 * 表达式索引与视图存在兼容性问题。通过视图进行查询时，无法使用上表达式索引。
 * 表达式索引与 Binding 存在兼容性问题。当表达式索引中的表达式存在常量时，对应查询所建的 Binding 会扩大范围。假设表达式索引中的表达式为 `a+1`，对应的查询条件为 `a+1 > 2`。则建立的 Binding 为 `a+? > ?`，这会导致像 `a+2 > 2` 这样的查询也会强制使用表达式索引，得到一个较差的执行计划。这同样影响 SQL Plan Management (SPM) 中的捕获和演进功能。
 * 多值索引写入的数据必须与定义类型完全匹配，否则数据写入失败。详见[创建多值索引](/sql-statements/sql-statement-create-index.md#创建多值索引)。
+* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```
Comment (qiancai) [inline sql-statements/sql-statement-create-index.md:411]:
```suggestion
* TiDB 对[分区表](/partitioned-table.md)进行了扩展。你可以指定 `GLOBAL` 索引选项将 `UNIQUE INDEX` 设置为[全局索引](/partitioned-table.md#全局索引)。该扩展与 MySQL 不兼容。
```

File: sql-statements/sql-statement-create-table.md [hunk 23]
```diff
@@ -234,6 +235,7 @@ mysql> DESC t1;
 * `COMMENT` 属性不支持 `WITH PARSER` 选项。
 * TiDB 在单个表中默认支持 1017 列，最大可支持 4096 列。InnoDB 中相应的数量限制为 1017 列，MySQL 中的硬限制为 4096 列。详情参阅 [TiDB 使用限制](/tidb-limitations.md)。
 * 分区表支持 `HASH`、`RANGE`、`LIST` 和 `KEY` [分区类型](/partitioned-table.md#分区类型)。对于不支持的分区类型，TiDB 会报 `Warning: Unsupported partition type %s, treat as normal table` 错误，其中 `%s` 为不支持的具体分区类型。
+* TiDB 对[全局索引](/partitioned-table.md#全局索引)进行了扩展。你可以使用 `GLOBAL` IndexOption 将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为全局索引。该扩展与 MySQL 不兼容。
```
Comment (qiancai) [inline sql-statements/sql-statement-create-table.md:238]:
```suggestion
* TiDB 对[分区表](/partitioned-table.md)进行了扩展。你可以指定 `GLOBAL` 索引选项将 `PRIMARY KEY` 或 `UNIQUE INDEX` 设置为[全局索引](/partitioned-table.md#全局索引)。该扩展与 MySQL 不兼容。
```

File: system-variables.md [hunk 24]
```diff
@@ -1919,13 +1919,17 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+> 该变量控制的功能为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
 - 类型：布尔型
 - 默认值：`OFF`
 - 可选值：`OFF`，`ON`
-- 这个变量用于控制是否支持对分区表创建 `Global index`。`Global index` 当前正处于开发阶段，**不推荐修改该变量值**。
+- 该变量控制是否支持为分区表创建[全局索引](/partitioned-table.md#全局索引)。启用此变量后，你可以使用 `GLOBAL` IndexOption 或 ColumnOption 创建不包含分区表达式中所有列的唯一索引。
```
Comment (qiancai) [inline system-variables.md:1932]:
```suggestion
- 该变量控制是否支持为分区表创建[全局索引](/partitioned-table.md#全局索引)。启用此变量后，你可以通过在索引定义中添加 `GLOBAL` 选项创建不包含分区表达式中所有列的唯一索引。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**